crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -21,24 +21,18 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
         let comments = f.context().comments().clone();
         let dangling_item_comments = comments.dangling_comments(item);
 
-        write!(
-            f,
-            [
-                text("case"),
-                space(),
-                format_with(|f: &mut PyFormatter| {
-                    let comments = f.context().comments();
-
-                    for comment in comments.leading_trailing_comments(pattern) {
-                        // This is a lie, but let's go with it.
-                        comment.mark_formatted();
-                    }
-
-                    // Replace the whole `format_with` with `pattern.format()` once pattern formatting is implemented.
-                    not_yet_implemented_custom_text("NOT_YET_IMPLEMENTED_Pattern", pattern).fmt(f)
-                }),
-            ]
-        )?;
+        write!(f, [text("case"), space()])?;
+        let leading_pattern_comments = comments.leading_comments(pattern);
+        if leading_pattern_comments.is_empty() {
+            pattern.format().fmt(f)?;
+        } else {
```

Comment:
Nit
```suggestion
        if comments.has_leading_comments(pattern) {
            pattern.format().fmt(f)?;
        } else {
```

Do we need to handle trailing own line comments too? I guess my question is. Does this implement the parenthesizing exactly like black or is the goal to unblock the work on match cases and we follow up with a matching formatting later?

Reply:
By trailing, do you mean the following?

```python
match foo:
    case (
        # test
    foo  # trailing 1
    # trailing 2
# trailing 3
    ):
        pass
```

If so, they're handled correctly:

```python
match foo:
    case (
        # test
        x as NOT_YET_IMPLEMENTED_PatternMatchAs  # trailing 1
        # trailing 2
        # trailing 3
    ):
        pass
```

Reply:
Oh, also this change isn't necessary as we need the leading comments to be added 😅

Reply:
Ah, you're correct, without the leading comments the parentheses aren't added.